examples/hardhat/hardhat.config.ts

```diff
@@ -0,0 +1,99 @@
+import { HardhatUserConfig, task } from "hardhat/config";
+// #region config-import
+import { assert } from "console";
+import "@oasisprotocol/sapphire-hardhat";
+// #endregion config-import
+import "@nomicfoundation/hardhat-toolbox";
+
+// Hardhat Node and sapphire-dev test key
+const firstPrivateKey = `0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80`;
+const accounts = process.env.PRIVATE_KEY
+  ? [process.env.PRIVATE_KEY]
+  : [firstPrivateKey];
+
+// #region config-tasks
+task("deploy").setAction(async (_args, hre) => {
+  const Vigil = await hre.ethers.getContractFactory("Vigil");
+  const vigil = await Vigil.deploy();
+  const vigilAddr = await vigil.waitForDeployment();
+
+  console.log(`Vigil address: ${vigilAddr.target}`);
+  return vigilAddr.target;
+});
+
+task("createSecret")
```

Comment:
Reading hardhat tasks documentation, I think they prefer dashes instead of the camel casing.

```suggestion
task("create-secret")
```